target/linux/ath79/dts/qca9557_dongwon_dw02-412h-128m.dts

```diff
@@ -0,0 +1,43 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qca9557_dongwon_dw02-412h.dtsi"
+
+/ {
+	model = "Dongwon T&I DW02-412H (128M)";
+	compatible = "dongwon,dw02-412h-128m", "qca,qca9557";
+};
+
+&nand {
+	status = "okay";
+
+	partitions {
+		compatible = "fixed-partitions";
+		#address-cells = <1>;
+		#size-cells = <1>;
+
+		partition@0 {
+			label = "current";
+			reg = <0x0 0x1000000>;
+			read-only;
+		};
+
+		partition@1000000 {
+			label = "firmware";
+			reg = <0x1000000 0x7000000>;
+
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "kernel";
+				reg = <0x0 0x800000>;
+			};
+
+			partition@800000 {
+				label = "ubi";
+				reg = <0x800000 0x6800000>;
+			};
+		};
```

Comment:
Does this nesting of firmware and kernel/ubi work? Why not just use kernel/ubi?

Reply:
It works well. I think this would be good for dumping all areas used by openwrt in NAND, as in the example of NBG6716 ([#](https://github.com/openwrt/openwrt/blob/20b6e014a61a9b40c30fab8184a9abd17f55085b/target/linux/ath79/dts/qca9558_zyxel_nbg6716.dts#L143)).

Reply:
I still don't think it's a good idea to have firmware and ubi partition. This whole nesting seems unnecessary to me.

Reply:
Okay, I'll revert to original state for this change.